CoreScriptsRoot/Modules/PlayerDropDown.lua

```diff
@@ -567,6 +571,56 @@ function createPlayerDropDown()
 	return playerDropDown
 end
 
+--- GetCore Blocked/Muted/Friended events.
+
+local readFlagSuccess, flagEnabled = pcall(function() return settings():GetFFlag("CorescriptGetCorePlayerEvents") end)
+local CorescriptPlayerEventsEnabled = readFlagSuccess and flagEnabled
+
+if CorescriptPlayerEventsEnabled then
+	local PlayerBlockedEvent = Instance.new("BindableEvent")
+	local PlayerUnblockedEvent = Instance.new("BindableEvent")
+	local PlayerMutedEvent = Instance.new("BindableEvent")
+	local PlayerUnMutedEvent = Instance.new("BindableEvent")
+	local PlayerFriendedEvent = Instance.new("BindableEvent")
+	local PlayerUnFriendedEvent = Instance.new("BindableEvent")
+
+	BlockStatusChanged:connect(function(userId, isBlocked)
+		local player = PlayersService:GetPlayerByUserId(userId)
+		if player then
+			if isBlocked then
+				PlayerBlockedEvent:Fire(player)
+			else
+				PlayerUnblockedEvent:Fire(player)
+			end
+		end
+	end)
+
+	MuteStatusChanged:connect(function(userId, isMuted)
+		local player = PlayersService:GetPlayerByUserId(userId)
+		if player then
+			if isMuted then
+				PlayerMutedEvent:Fire(player)
+			else
+				PlayerUnMutedEvent:Fire(player)
+			end
+		end
+	end)
+
+	LocalPlayer.FriendStatusChanged:connect(function(player, friendStatus)
+		if friendStatus == Enum.FriendStatus.Friend then
+			PlayerFriendedEvent:Fire(player)
+		elseif friendStatus == Enum.FriendStatus.NotFriend then
+			PlayerUnFriendedEvent:Fire(player)
+		end
+	end)
+
+	StarterGui:RegisterGetCore("PlayerBlockedEvent", function() return PlayerBlockedEvent end)
```

Comment:
These could probably just return the BindableEvent.Event properties so that developers can't try to move/fire the event objects.